tools/visual-tester/src/runner.js

```diff
@@ -195,8 +236,8 @@ module.exports = class Runner {
      */
     try {
       /* Create new page and navigate to it. */
-      const page = await this.puppet.newPage();
-      await page.goto(`${this.config.baseUrl}${url}`, {
+      const page = this.config.disableIncognito ? await this.incognitoContext.newPage() : await this.puppet.newPage();
```

Comment:
Why do we need to have `incognitoContext` as part of `this`

and probably is time to extract this to a function ? maybe